src/service/middleware/retry/mod.rs

```diff
@@ -0,0 +1,55 @@
+use futures_util::future;
+use http::{Request, Response};
+use hyper::{Body, Error};
+use tower::retry::Policy;
+
+#[derive(Clone)]
+pub enum RetryConfig {
+    None,
+    Simple(usize),
+}
+
+impl Policy<Request<String>, Response<hyper::Body>, hyper::Error> for RetryConfig {
+    type Future = futures_util::future::Ready<Self>;
+
+    fn retry(
+        &self,
+        _req: &Request<String>,
+        result: Result<&Response<Body>, &Error>,
+    ) -> Option<Self::Future> {
+        match self {
+            RetryConfig::None => None,
+            RetryConfig::Simple(count) => match result {
+                Ok(_) => None,
+                Err(_) => {
+                    if *count > 0 {
+                        Some(future::ready(RetryConfig::Simple(count - 1)))
+                    } else {
+                        None
+                    }
+                }
+            },
```

Comment:
It'd be nice, but not strictly a blocker, if we could somehow pass the retry attempt number to the tracing layer so we can include it as a field (`http.resend_count`) on the API call span. Ref: https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/http/#http-client

Reply:
good point. I am wondering what the best way would be to pass that information down? Currently retry layer wraps tracing layer. We could just modify the request header to add the retry attempt, and then have it removed on tracing layer(seems kinda ugly to me)? Alternatively, we can have tracing layer wrap retry layer, and let retry layer create its own span with `http.resend_count`


Also, it seems like there is a rename planned for `http.resend_count`, so we might want to have this implemented after?: 
https://github.com/open-telemetry/opentelemetry-specification/issues/3182